### PR TITLE
[Core] parsing integers instead of bools in pybind lists because ... reasons

### LINKED ISSUE #8163: How to add CalculateOnIntegrationPoints for bool variables to Python?
**Description**
Apparently CalculateOnIntegrationPoints for boolean variables is not exported to python
```
TypeError: CalculateOnIntegrationPoints(): incompatible function arguments. The following argument types are supported:
    1. (self: Kratos.Element, arg0: Kratos.IntegerVariable, arg1: Kratos::ProcessInfo) -> list
    2. (self: Kratos.Element, arg0: Kratos.DoubleVariable, arg1: Kratos::ProcessInfo) -> list
    3. (self: Kratos.Element, arg0: Kratos.Array1DVariable3, arg1: Kratos::ProcessInfo) -> list
    4. (self: Kratos.Element, arg0: Kratos.VectorVariable, arg1: Kratos::ProcessInfo) -> list
    5. (self: Kratos.Element, arg0: Kratos.MatrixVariable, arg1: Kratos::ProcessInfo) -> list

Invoked with: <Kratos.Element object at 0x7fc25ec657f0>, <Kratos.BoolVariable object at 0x7fc25b35eb70>, <Kratos.ProcessInfo object at 0x7fc25ec4e570>
```

I tried adding
`.def("CalculateOnIntegrationPoints", CalculateOnIntegrationPoints<Element, bool>)`
to line 465ish of `.../kratos/python/add_mesh_to_python.cpp`, but it seems not to be enough:
`SystemError: <built-in method CalculateOnIntegrationPoints of PyCapsule object at 0x7f65df530060> returned a result with an error set`

Is it missing something else?

**Scope**
- Python

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
Fixes #8163 (again)

**Changelog**
- Correcting typo in solid element
- Filling of Zero in the mesh elements (default core ones), in order to create a test
- Parsing booleans as integers
- Added test

### CODE REVIEW

File: kratos/tests/test_integration_points.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        mp.CreateNewElement("Element2D4N", 2, [6,4,7,8], mp.GetProperties()[1])
+        mp.CreateNewElement("Element2D4N", 3, [1,2,4,6], mp.GetProperties()[1])
+        mp.CreateNewElement("Element2D4N", 4, [4,2,3,7], mp.GetProperties()[1])
+        mp.CreateNewElement("Element2D4N", 5, [2,1,5,3], mp.GetProperties()[1])
+
+        for elem in mp.Elements:
+            # Boolean
+            out = elem.CalculateOnIntegrationPoints(KratosMultiphysics.IS_RESTARTED, mp.ProcessInfo)
+            for value in out:
+                self.assertEqual(value,False)
```
Comment (philbucher) [inline kratos/tests/test_integration_points.py]:
```suggestion
                self.assertFalse(value)

```

File: kratos/python/add_mesh_to_python.cpp [hunk 2]
```diff
@@ -147,13 +147,25 @@ pybind11::list CalculateOnIntegrationPoints(
...
 
+template< class TObject>
+pybind11::list CalculateOnIntegrationPointsBool(
+    TObject& dummy, const Variable<bool>& rVariable, const ProcessInfo& rProcessInfo)
+{
+    std::vector<bool> Output;
+    dummy.CalculateOnIntegrationPoints(rVariable, Output, rProcessInfo);
+    pybind11::list result;
+    for (std::size_t j = 0; j < Output.size(); j++) {
+        result.append(static_cast<int>(Output[j]));
```
Comment (philbucher) [inline kratos/python/add_mesh_to_python.cpp:164]:
```suggestion
        result.append(static_cast<bool>(Output[j]));
```

why not like this?

**Review by philbucher — COMMENTED** [review]
minor otherwise cool

File: kratos/python/add_mesh_to_python.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/python/add_mesh_to_python.cpp:164]: Its already bool, the problem is the pybind11::list

File: kratos/tests/test_integration_points.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                self.assertFalse(value)
+
+            # Int
+            out = elem.CalculateOnIntegrationPoints(KratosMultiphysics.LOAD_RESTART, mp.ProcessInfo)
+            for value in out:
+                self.assertEqual(value,0)
+            # Double
+            out = elem.CalculateOnIntegrationPoints(KratosMultiphysics.PRESSURE, mp.ProcessInfo)
+            for value in out:
+                self.assertEqual(value,0.0)
```
Comment (philbucher) [inline kratos/tests/test_integration_points.py]:
```suggestion
                self.assertAlmostEqual(value,0.0)

```

File: kratos/tests/test_integration_points.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            # Array
+            out = elem.CalculateOnIntegrationPoints(KratosMultiphysics.DISPLACEMENT, mp.ProcessInfo)
+            for value in out:
+                self.assertEqual(value[0],0.0)
+                self.assertEqual(value[1],0.0)
+                self.assertEqual(value[2],0.0)
```
Comment (philbucher) [inline kratos/tests/test_integration_points.py]:
```suggestion
                self.assertAlmostEqual(value[0],0.0)
                self.assertAlmostEqual(value[1],0.0)
                self.assertAlmostEqual(value[2],0.0)

```

**Review by philbucher — COMMENTED** [review]
minor then I will approve, sorry didn't see the double comparison before

**Review by philbucher — APPROVED** [review]

**Comment by marandra** [comment]
I can confirm that my cases are working now.
Thanks @loumalouomega !

**Review by marandra — APPROVED** [review]

**Pull request merged**